### ISSUE TITLE
[core] Resource table is being parsed twice

### ISSUE THREAD

**Issue by pubiqq** [skylot/jadx#2263]
### Issue details

This bug used to only increase parsing time, but now because of it and b85900a, resources are being generated with double entries.

<details><summary>Screenshot</summary>
<p>

![](https://github.com/user-attachments/assets/29571ea2-1b85-440d-83a0-302fd6f11aaa)

</p>
</details> 

### Jadx version

[`cca706c`](https://github.com/skylot/jadx/tree/cca706c94f31b823885aa863366510dffebdb932)

**Comment by skylot** [comment]
Fixed. Some details:
- actually running decode of arsc file twice is fine. First time needed to collect some info used during decompilation (like mapping to replace resource IDs with res names in code). And second time is lazy loading/generation of resource xml files. So second time is optional (not invoked in jadx-gui if user not open arsc file content).
- issue was introduced in PR #2165 by keeping parser instance in memory like a singleton, so second decode was adding arsc entries again.
- I changed code to not store parser instance and also, just in case, to create entries storage right before decode to prevent possible duplication.

@pubiqq thank you for report, it was very helpful :+1:
